### PR TITLE
Use preferred package header for generated code

### PR DESCRIPTION
**Pull request by jdbaldry**
> To convey to humans and machine tools that code is generated, generated source should have a line that matches the following regular expression (in Go syntax):
> `^// Code generated .* DO NOT EDIT\.$`

https://pkg.go.dev/cmd/go#hdr-Generate_Go_files_by_processing_source

*Note:* I realize that this generation affects more than just Go files so perhaps this change is undesirable.

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

### CODE REVIEW

**Comment by joanlopez** [comment]
> _Note:_ I realize that this generation affects more than just Go files so perhaps this change is undesirable.

LGTM although I guess that's a better question for @sdboyer

File: pkg/codegen/tmpl/gen_header.tmpl [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// THIS FILE IS GENERATED. EDITING IS FUTILE.
+// Code generated. DO NOT EDIT.
```
Comment (sdboyer) [inline pkg/codegen/tmpl/gen_header.tmpl:1]: This doesn't actually match the regexp - there needs to be a space after `generated`.

```suggestion
// Code generated - EDITING IS FUTILE. DO NOT EDIT.
```

**Review by sdboyer — CHANGES_REQUESTED** [review]
i discussed this with @sakjur recently, and am good with this. Just need a pattern that satisfies the regexp

File: pkg/codegen/tmpl/gen_header.tmpl (hunk 1 above)
Comment (jdbaldry) [reply pkg/codegen/tmpl/gen_header.tmpl:1]: Good catch, thanks for checking thoroughly!

I'll make the proposed changes, and regen.

**Review by sdboyer — APPROVED** [review]

**Comment by sdboyer** [comment]
LGTM now - if we have concerns about how this looks in other file types, we can always parameterize the template

**Comment by jdbaldry** [comment]
Thanks for the quick reviews. This kind of PRs can become frustrating when left outstanding for too long :)

**Pull request merged**